models/User.go

```diff
@@ -1143,11 +1143,11 @@ func getTaxForBiddingUser(tx *gorm.DB, stockId uint32, stockQuantity uint64, sto
 	var tax uint64
 
 	transactionSummary := &TransactionSummary{UserId: userId, StockId: stockId}
-	tx.First(&transactionSummary)
+	tx.Where("userId = ? AND stockId = ?", userId, stockId).First(&transactionSummary)
```

Comment:
Why was this changed? Just regarding previous PR why is Error not handelled here?

Reply:
Previously, the Primary key was (userID, stockID). So `tx.First(&transactionSummary)` will work because the userID and stockID fields of the transactionSummary object are populated.

Now, the Primary key is ID, which is an autoIncrementing field. And so in order to retrieve from the database, we need to specify Where clauses since (userId, stockID) are no longer the Primary Key